iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -330,29 +281,7 @@ public override async Task CompleteAsync(string lockToken, CancellationToken can
             Task completeOperationCompletion;
             lock (this.syncRoot)
             {
-                if (!lockToken.StartsWith(this.generationId))
-                {
-                    throw new IotHubException(
-                        "Lock token is stale or never existed. The message will be redelivered, please discard this lock token and do not retry operation.",
-                        isTransient:false);
-                }
-
-                if (this.completionQueue.Count == 0)
-                {
-                    throw new IotHubException("Unknown lock token.", isTransient:false);
-                }
-
-                string actualLockToken = this.completionQueue.Peek();
-                if (lockToken.IndexOf(actualLockToken, GenerationPrefixLength, StringComparison.Ordinal) != GenerationPrefixLength ||
-                    lockToken.Length != actualLockToken.Length + GenerationPrefixLength)
-                {
-                    throw new IotHubException(
-                        $"Client MUST send PUBACK packets in the order in which the corresponding PUBLISH packets were received (QoS 1 messages) per [MQTT-4.6.0-2]. Expected lock token: '{actualLockToken}'; actual lock token: '{lockToken}'.",
-                        isTransient:false);
-                }
-
-                this.completionQueue.Dequeue();
-                completeOperationCompletion = this.channel.WriteAndFlushAsync(actualLockToken);
+                completeOperationCompletion = this.channel.WriteAndFlushAsync(lockToken);
```

Comment:
This change requires more test to be added. Add test or hold on this change.

Reply:
Reverted. There is more code coupling together. Will leave as it is.